File: gluecksrad-app/src/app/shared/model/gamestate.ts
import {Question} from "./question";
import {Sentence} from "./sentence";
import {Category} from "./category";
import {PlayMode} from "./playmode.enum";

export interface Gamestate {
  score?: number;
  lives?: number;
  roundsPlayed?: number;
  isWon?: boolean;
  nothingLeftToGuess?: boolean;
  currentMode?: PlayMode
  currentQuestion?: Question;
  currentSentence?: Sentence;
  category?: Category;
}

File: accessing-data-mysql/src/main/java/ch/bbbaden/gluecksrad/api/ToolEntities/Gamestate.java
package ch.bbbaden.gluecksrad.api.ToolEntities;

import ch.bbbaden.gluecksrad.model.CategoryEntity;
import ch.bbbaden.gluecksrad.model.QuestionEntity;
import ch.bbbaden.gluecksrad.model.SentenceEntity;

public class Gamestate {

    public int score = 0;
    public int lives = 3;
    public int roundsPlayed = 0;
    public boolean isWon = false;
    public boolean nothingLeftToGuess = false;
    public PlayMode currentMode;
    public QuestionEntity currentQuestion;
    public SentenceEntity currentSentence;
    public CategoryEntity category;
}
